src/Umbraco.Core/Packaging/PackageDataInstallation.cs

```diff
@@ -692,10 +693,45 @@ private IContentType UpdateContentTypeFromXml(XElement documentType, IContentTyp
             UpdateContentTypesAllowedTemplates(contentType, infoElement.Element("AllowedTemplates"), defaultTemplateElement);
             UpdateContentTypesPropertyGroups(contentType, documentType.Element("Tabs"));
             UpdateContentTypesProperties(contentType, documentType.Element("GenericProperties"));
+            UpdateHistoryCleanupPolicy(contentType, documentType.Element("HistoryCleanupPolicy"));
 
             return contentType;
         }
 
+        private void UpdateHistoryCleanupPolicy(IContentType contentType, XElement element)
+        {
+            contentType.HistoryCleanup ??= new HistoryCleanup();
+
+            if (element == null)
+            {
+                contentType.HistoryCleanup = new HistoryCleanup();
+                return;
+            }
+
+            if (bool.TryParse(element.Attribute("preventCleanup")?.Value, out var preventCleanup))
+            {
+                contentType.HistoryCleanup.PreventCleanup = preventCleanup;
+            }
+
+            if (int.TryParse(element.Attribute("keepAllVersionsNewerThanDays")?.Value, out var keepAll))
+            {
+                contentType.HistoryCleanup.KeepAllVersionsNewerThanDays = keepAll;
+            }
+            else
+            {
+                contentType.HistoryCleanup.KeepAllVersionsNewerThanDays = null;
+            }
+
+            if (int.TryParse(element.Attribute("keepLatestVersionPerDayForDays")?.Value, out var keepLatest))
+            {
+                contentType.HistoryCleanup.KeepLatestVersionPerDayForDays = keepLatest;
+            }
+            else
+            {
+                contentType.HistoryCleanup.KeepLatestVersionPerDayForDays = null;
+            }
```

Comment:
Also keep the existing value when this setting isn't provided/valid:
```suggestion
```

Reply:
What if you intentionally want to remove the value?